crates/app/src/lib.rs

```diff
@@ -7,22 +7,21 @@
 #![deny(missing_docs)]
 
 mod host_component;
-pub mod locked;
-mod metadata;
-pub mod values;
+pub use spin_locked_app::locked;
+pub use spin_locked_app::values;
+pub use spin_locked_app::{Error, MetadataKey, Result};
```

Comment:
Do we need these to be `pub`?

Reply:
The intent here was to re-export the things that `spin-app` previously exported, so that other crates didn't need to import both and change a bunch of references.  (My previous crack at this did _not_ re-export, and resulted in niggly dependency changes throughout the codebase, which is why I went this route this time.)